doc/src/examples/dot_product

```diff
@@ -0,0 +1,41 @@
+// Calculate the dot product of two points
+
+#include <iostream>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/arithmetic/dot_product.hpp>
+#include <boost/geometry/geometries/adapted/boost_array.hpp>
+
+namespace bg = boost::geometry; /*< Convenient namespace alias >*/
+
+BOOST_GEOMETRY_REGISTER_BOOST_ARRAY_CS(cs::cartesian)
+
+
+int main()
+{
+
+	double dp1,dp2,dp3,dp4;
+	bg::model::point<double, 3, bg::cs::cartesian> point1(1.0, 2.0, 3.0);
+	bg::model::point<double, 3, bg::cs::cartesian> point2(4.0, 5.0, 6.0);
+ 
+	dp1 = bg::dot_product(point1, point2);
+ 
+	std::cout << "Dot Product 1: "<< dp1 << std::endl; 
+	
+	bg::model::point<double, 2, bg::cs::cartesian> point3(3.0, 2.0);
+	bg::model::point<double, 2, bg::cs::cartesian> point4(4.0, 7.0);
+	
+	dp2 = bg::dot_product(point3, point4);
+	
+ 	std::cout << "Dot Product 2: "<< dp2 << std::endl; 
+ 	
+ 	boost::array<double, 2> a =  {1, 2};
+  boost::array<double, 2> b =  {2, 3};
```

Comment:
tab is missing

Reply:
More precisely, 4 spaces as per https://github.com/boostorg/geometry/wiki/Guidelines-for-Developers#code-formatting-and-indentation